### PR TITLE
World

### PR DESCRIPTION
**Pull request by theraven262**
Pretty much everything I had planned for this branch is in:

- [x] Skygen integration
- [x] Hell Biome
- [x] Underground Expansion to -8000 (the underground biome is down to -1024 now)
- [x] Skygen sky biome override, for the spawn and sky-cities (starts at 1024 but is configurable)
- [x] Actual bottom world-boundary
// The "content" stuff that came along as I was developing:
- [x] Checkerboard variations, new ones can also be easily registered now
- [x] Bonsai trees, basically saplings rework
- [x] All tree types can now be grown from sapling
- [x] Cherry tree added to snowy grassland
- [x] Patinated copper added

### CODE REVIEW

File: mods/flora/functions.lua [hunk 1]
```diff
@@ -72,14 +76,42 @@ function flora.grow_sapling(pos)
...
+		minetest.pos_to_string(pos))
+		flora.grow_new_big_acacia_tree(pos)
 	elseif node.name == "flora:aspen_sapling" then
 		minetest.log("action", "An aspen sapling grows into a tree at "..
 			minetest.pos_to_string(pos))
 		flora.grow_new_aspen_tree(pos)
+	elseif node.name == "flora:aspen_sapling_big" then
+		minetest.log("action", "A big aspen sapling grows into a huge aspen tree at "..
+		minetest.pos_to_string(pos))
+		flora.grow_new_big_aspen_tree(pos)
```
Comment (wrynvera) [inline mods/flora/functions.lua]: Can you combine this into a standardized function taking the schem parameter + log message parameter, as well as position parameters.

**Comment by wrynvera** [comment]
On a brief scan through, everything else seems alright.
I am worried about the implications of two diverging skygen however, and how hard it might get to maintain it.

File: mods/flora/functions.lua (hunk 1 above)
Comment (theraven262) [reply mods/flora/functions.lua]: Should I though? The big jungle tree naming scheme would be a big problem, and if we attempted to change it that would percolate across a lot of code. On the other hand, having a table containing all the sapling strings + tree names + function names would not make it any more readable.

**Comment by theraven262** [comment]
> On a brief scan through, everything else seems alright. I am worried about the implications of two diverging skygen however, and how hard it might get to maintain it.

Not at all, our skygen diverges from the main one by two checks in the main loop, I can basically copy-paste the whole code to my repo.

**Comment by wrynvera** [comment]
Can we split that into a check on skygen main then, and use a submodule to connect them?
That is more efficient than maintaining two versions.

File: mods/flora/functions.lua (hunk 1 above)
Comment (wrynvera) [reply mods/flora/functions.lua]: `function_grow_tree(tree_name, description, sap_x, sap_y, sap_z)`

**Comment by theraven262** [comment]
If i can somehow put skyboxes on our repo, but not on the main one. The skyboxes take a lot of space, and I don't want the mod download speed to suffer.

**Comment by wrynvera** [comment]
You can split the textures into a separate repo I suppose, and connect both those with submodules.
I am not for maintaining two separate versions, because even if it is copy and paste, at some point, from experience, they might diverge and maintenance will be an issue.

**Comment by theraven262** [comment]
Done. Skygen is now linked as a submodule (that was much easier than i expected, nice) and its skyboxes are another module inside it. Sorry that took a while, I found a bug in skygen that took a bit of time to fix.

**Pull request merged**